### PR TITLE
Added link to the resources sidebar from plot module documentation

### LINKED ISSUE #5603: Add link to the resources sidebar from plot module documentation
**Description of the desired feature**

In the caption to the figure showing custom symbols, there is a reference to "RESOURCES page (see sidebar)". This text should be replaced with an actual relative URL.  This is in the [plot](https://docs.generic-mapping-tools.org/dev/plot.html#s) module.

### PR DESCRIPTION
**Pull request by soham4abc**
Signed-off-by: Soham Banerjee <sohambanerjee4abc@hotmail.com>

**Description of proposed changes**
Added link `<users-contrib-symbols>` to the resources sidebar from plot module documentation.

Fixes #5603 

**Reminders**

- [x] Make sure that your code follows our style. Use the other functions/files as a basis.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Describe changes to function behavior and arguments in a comment below the function declaration.
- [x] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by soham4abc** [comment]
@meghanrjones  PTAL

File: doc/rst/source/explain_symbols.rst_ [hunk 1]
```diff
@@ -293,7 +293,7 @@
         Custom symbols are designed using the :ref:`Custom Symbol Macro Language <Custom_symbols>`.
         We supply about 40 custom symbols, and users have contributed discipline-specific
         symbols for structural geology and marine biology that you can explore from
-        the RESOURCES page (see sidebar), so you have lots of
+        the :ref:`RESOURCES page (see sidebar) <symbols_gallery>`, so you have lots of
```
Comment (meghanrjones) [inline doc/rst/source/explain_symbols.rst_:296]:
```suggestion
        the :ref:`RESOURCES page <symbols_gallery>`, so you have lots of
```

**Review by meghanrjones — COMMENTED** [review]
Thanks!

Just one tiny suggestion, since with the new link the reader will no longer need to navigate through the sidebar.

**Comment by soham4abc** [comment]
Sure.. I made the changes. Please do have a look! @meghanrjones

**Comment by soham4abc** [comment]
Does this require any changes @meghanrjones ??

**Review by meghanrjones — APPROVED** [review]
Looks good, thanks!

**Pull request merged**